### PR TITLE
Allow customization of rendered PDF

### PR DESCRIPTION
**Pull request by chrisortman**
See discussion in #783 

This adds an extension point that can be used to customize the header & footer of the generated PDF.

I also included some changes to help with linting the cocoa pod, but I can rebase those away if needed.

### CODE REVIEW

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
+//  ORKHTMLPDFPageRenderer.h
+//  ResearchKit
+//
+//  Created by Ortman, Chris E on 8/10/16.
+//  Copyright © 2016 researchkit.org. All rights reserved.
+//
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.h]: Update to RK header with your copyright.

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.m [hunk 2]
```diff
@@ -0,0 +1,41 @@
+//
+//  ORKHTMLPDFPageRenderer.m
+//  ResearchKit
+//
+//  Created by Ortman, Chris E on 8/10/16.
+//  Copyright © 2016 researchkit.org. All rights reserved.
+//
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.m]: Update to RK header with your copyright.

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.m [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+#import "ORKErrors.h"
+#import "ORKHelpers_Internal.h"
+#import "ORKDefines.h"
+#import "ORKHTMLPDFPageRenderer.h"
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.m]: Add extra new line.

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+#import <CoreGraphics/CoreGraphics.h>
+#import <UIKit/UIKit.h>
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.h]: Add extra new line.

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.h [hunk 5]
```diff
@@ -0,0 +1,19 @@
+//
+//  ORKHTMLPDFPageRenderer.h
+//  ResearchKit
+//
+//  Created by Ortman, Chris E on 8/10/16.
+//  Copyright © 2016 researchkit.org. All rights reserved.
+//
+
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.h]: 2 new lines before and after import block as per RK Coding Guidlines.

Add extra new line.

**Comment by umerkhan-apple** [comment]
@chrisortman Is okay to close this PR for now and re-open when you are able to make changes and resolve the merge conflicts?

**Comment by chrisortman** [comment]
Can you leave it? I'm on vacation right now but can probably get to it this
weekend

**Comment by umerkhan-apple** [comment]
@chrisortman Sounds good! Enjoy your vacation.

**Comment by chrisortman** [comment]
@umerkhan-apple Here you go, let me know if anything else I need to do.

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.h [hunk 6]
```diff
@@ -0,0 +1,43 @@
+/*
+ Copyright (c) 2016, Chris Ortman, University of Iowa, All rights reserved.
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.h]: 2017

File: ResearchKit/Common/ORKHTMLPDFPageRenderer.m [hunk 7]
```diff
@@ -0,0 +1,65 @@
+/*
+ Copyright (c) 2016, Chris Ortman, University of Iowa, All rights reserved.
```
Comment (umerkhan-apple) [inline ResearchKit/Common/ORKHTMLPDFPageRenderer.m]: 2017

**Comment by chrisortman** [comment]
Here you go @umerkhan-apple , sorry it took me so long to get back to this.

**Comment by scdi** [comment]
How could one implement the changes now?

Also, in the code below, I am trying to get signature, name, date applied to the PDF rendered.
I can only get those rendered if I use the html in the consent steps. When I try to use an html file independent of the consent step, the signature page does not appear.

//VisualConsentStep
    let consentDocument = ConsentDocument
    
    let visualConsentStep = ORKVisualConsentStep(identifier: "VisualConsentStep", document: consentDocument)
    
    steps += [visualConsentStep]
    
    /*//Remove birthdate question from this step 
    //Birthdate is now moved to demography
    
    
    */
    
    //ConsentReviewStep
    let signature = consentDocument.signatures!.first!
    for anything in consentDocument.signatures! {
        print("My Signatures: \(anything.identifier)")
    }
    
    let reviewConsentStep = ORKConsentReviewStep(identifier: "ConsentReviewStep", signature: signature, inDocument: consentDocument)
    
    reviewConsentStep.text = "Review Consent!"
    reviewConsentStep.title = "Review Completed"
    reviewConsentStep.reasonForConsent = "I have read the informed consent, and questions about my participation have been answered to my satisfaction. By pressing this \"Agree\" button, I agree to participate in this research study. I understand that my participation is entirely voluntary."
   
 let url = NSURL(fileURLWithPath: NSBundle.mainBundle().pathForResource("eConsentWebPage", ofType: "html")!)
    
    do {
     
        let text = try String(contentsOfURL: url)
        reviewConsentStep.consentDocument.htmlReviewContent = text
        
     } catch {
        print("Something went wrong!")
    }


Is there a way to append the signature page to an HTML file representing a consent form?

**Comment by chrisortman** [comment]
@scdi I think that when you modify the htmlReviewContent on the step you have to provide your own way of rendering the signature. Otherwise it'd be tricky to figure out how to modify your HTML to insert an image somewhere.

If you are asking how you can make use of this change prior to it being merged you can either build from my branch or reference it in your pod file 

```
 pod 'ResearchKit', :git => 'https://github.com/chrisortman/ResearchKit.git', :branch => 'mcvrs'
```

**Comment by scdi** [comment]
Thank you so much.
J

**Comment by chrisortman** [comment]
@umerkhan-apple Anything else I can do to help this get merged?

**Comment by srinathtm-apple** [comment]
Hey @chrisortman , do you want to either update or remove the .swift-version file ?

**Comment by chrisortman** [comment]
@srinathtm-apple done

**Comment by srinathtm-apple** [comment]
@chrisortman merged. Thank so you much for this PR, this is a really good enhancement to the framework.

**Pull request merged**